messages/src/lib.rs

```diff
@@ -469,26 +498,25 @@ impl MacAddrs {
     /// that the full range is valid. Specifically, we check that the base MAC
     /// is not a multicast address, that all addresses in the range share the
     /// same OUI, and that count/stride are nonzero.
-    pub const fn new(base_mac: [u8; 6], count: u16, stride: u8) -> Option<Self> {
-        if Self::is_valid(base_mac, count, stride) {
-            Some(Self {
+    pub const fn new(base_mac: [u8; 6], count: u16, stride: u8) -> Result<Self, Error> {
```

Comment:
API musings: since this function will only ever return `Error::BadMacAddrBlock(...)`, maybe have it return the inner `BadMacAddrReason` directly (and let the caller wrap it if desired)?

Reply:
That maybe seems fine? Talking through that though, the `transceivers` task would need to wrap it immediately in order to send it to the host, since it has to be stuffed into an `SpResponse::Error` variant. On the host-side we're just going to deserialize it directly, so I don't think this code even runs, right? So I guess it's just up to you whether you'd like include that (minimal) logic in the Hubris task.

Reply:
I've reworked this so the `Error::BadMacAddrRange` variant now contains the reason _and_ the offending data. So the simplification here would require adding a new type to store both, and then using that in the `Error` variant.

Reply:
Okay, that seems good to me.